File: mobile/src/main/java/net/stecky/puzzlemethis/events/behaviors/BehaviorBase.java
package net.stecky.puzzlemethis.events.behaviors;

import net.stecky.puzzlemethis.events.Event;
import net.stecky.puzzlemethis.events.EventType;

import java.time.ZonedDateTime;

/**
 * Created by sb185219 on 11/8/2017.
 */

public abstract class BehaviorBase extends Event
{
    private String approach;

    public BehaviorBase(BehaviorSubType type)
    {
        super(type.isGood() == true ? EventType.GOOD_BEHAVIOR : EventType.BAD_BEHAVIOR);
    }

    public BehaviorBase(BehaviorSubType type, ZonedDateTime timestamp, String comments)
    {
        super(type.isGood() == true ? EventType.GOOD_BEHAVIOR : EventType.BAD_BEHAVIOR, type, timestamp, comments);
    }

    public String getApproach()
    {
        return approach;
    }

    public void setApproach(String approach)
    {
        this.approach = approach;
    }
}

File: mobile/src/main/java/net/stecky/puzzlemethis/events/behaviors/BehaviorBase.kt
package net.stecky.puzzlemethis.events.behaviors

import net.stecky.puzzlemethis.Person
import net.stecky.puzzlemethis.events.Event
import net.stecky.puzzlemethis.events.EventSubType
import net.stecky.puzzlemethis.events.EventType
import net.stecky.puzzlemethis.events.notes.Note
import java.time.Period
import java.time.ZonedDateTime

/**
 * Created by sb185219 on 11/14/2017.
 */
abstract class BehaviorBase(subject: Person, author: Person, type: EventType, subType: EventSubType,
                            timeStamp: ZonedDateTime, notes: List<Note>, var duration: Period) :
        Event(subject, author, type, subType, timeStamp, notes)
{}